remote/internal/metrics.go

```diff
@@ -33,38 +43,42 @@ type PendulumEntry struct {
 //
 // Returns:
 // - A slice of PendulumMetric structs containing the aggregated metrics.
-func AggregatePendulumMetrics(data [][]string, timeout_len float64, rangeType string) []PendulumMetric {
-	// get number of columns
-	n := len(data[0])
-
+func AggregatePendulumMetrics(
+	data [][]string,
+	timeout_len float64,
+	rangeType string,
+	sections []interface{}) []PendulumMetric {
 	// create waitgroup
 	var wg sync.WaitGroup
 
 	// create buffered channel to store results and avoid deadlock in main
-	res := make(chan PendulumMetric, n-2)
+	res := make(chan PendulumMetric, len(sections))
 
-	// iterate through each metric column (except for 'active' and 'time') and create goroutine for each
-	for m := 1; m < n-1; m++ {
+	// iterate through each metric column as specified in Sections config
+	// and create goroutine for each
+	for _, colName := range sections {
 		wg.Add(1)
 		go func(m int) {
 			defer wg.Done()
 			aggregatePendulumMetric(data, m, timeout_len, rangeType, res)
-		}(m)
+		}(csvColumns[colName.(string)])
 	}
 
-	// handle waitgroup in separate goroutine to allow main routine to process results as they become available.
+	// handle waitgroup in separate goroutine to allow main routine
+	// to process results as they become available.
 	var cleanup_wg sync.WaitGroup
 	cleanup_wg.Add(1)
+
 	go func() {
 		wg.Wait()
 		close(res)
 		cleanup_wg.Done()
 	}()
 
 	// deal with results
-	out := make([]PendulumMetric, n-2)
+	out := make([]PendulumMetric, len(data[0]))
 	for r := range res {
-		out[r.Index-1] = r
+		out[r.Index] = r
```

Comment:
I'd have to test this, but I used n-2 here (len(data[0]) - 2) to not create array entries for the 'active' and 'time' columns, so I'm not sure these changes will work as intended.

Reply:
![Screenshot from 2024-10-19 07-16-44](https://github.com/user-attachments/assets/6c767c74-c040-4ebd-82e8-1731b192f673)

Given this config

```
    opts = {
        ....
        sections = {
            "file",
            "filetype",
            "project",
            "directories",
        },
    },
```